src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -231,7 +233,9 @@
   (doseq [app apps]
     (let [{:keys [lifecycle-channel]} @(a/app-context app)
           restart-fn #(a/restart app)]
-      (async/>!! lifecycle-channel restart-fn))))
+      (when-not (async/offer! lifecycle-channel restart-fn)
+        (log/warnf "Too many SIGHUP restart requests queued (%s); ignoring!"
```

Comment:
If someone has log level at `warn` or `info` then they won't see the message from https://github.com/puppetlabs/trapperkeeper/pull/222/commits/9c017b830ffe86415c6d329d19262d112e4f01b4#diff-452afe9e02bf063094d280de34cf34bcR232 that says it's gotten a SIGHUP. If that is the case, then there isn't really an antecedent to `ignoring!` saying what it is ignoring. Might be better to say `ignoring new SIGHUP restart request` or something.